oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js

```diff
@@ -81,6 +91,92 @@ export function createFormAppInit({
   }
 }
 
+export const getFieldData = (uiMetadata, fieldPathPrefix = "") => {
+  return ({
+    fieldPath,
+    icon = "pencil",
+    fullLabelClassName,
+    compactLabelClassName,
+  }) => {
+    const fieldPathWithPrefix = fieldPathPrefix
+      ? `${fieldPathPrefix}.${fieldPath}`
+      : fieldPath;
+    // handling labels, always I take result of i18next.t if we get metadata/smth, we use it to debug
+    // help and hint if result is same as the key, don't render, if it is different render
+    const path = transformPath(fieldPathWithPrefix);
+    const {
+      help: modelHelp,
+      label: modelLabel,
+      hint: modelHint,
+      required,
+    } = _get(uiMetadata, path);
+    const label = i18next.t(modelLabel);
+    const help =
+      i18next.t(modelHelp) === modelHelp ? null : i18next.t(modelHelp);
+    const hint =
+      i18next.t(modelHint) === modelHint ? null : i18next.t(modelHint);
+    // full representation meaning jsx or small space representation (label with helptext in popup)
+    // text only without react elements
+    return {
+      fullRepresentation: {
```

Comment:
I would just return data needed and leave it to components calling this function, to render label component they want to.

Reply:
Currently this returns 3 different options, creating 3 different label components, where only one of them gets used every time